### ISSUE TITLE
Improve pkgin output when trying to install non-existent package

### ISSUE THREAD

**Issue by rmustacc** [TritonDataCenter/pkgsrc#143]
I created a fresh 2018q2 base 64 zone (18.2.0). When I started there I tried to install build essential and got some confusing results. The first time I tried to install it, it said nothing was available. I asked a second time and nothing was available. I then did a pkgin ls to check, certainly none of it was installed. I asked a final time and now it thinks it can be installed. Transcript follows:

```
[root@cross-dev ~]# pkgin in build-essentiail
reading local summary...
processing local summary...
processing remote summary (https://pkgsrc.joyent.com/packages/SmartOS/2018Q2/x86_64/All)...
pkg_summary.xz                                                                                                   100% 2238KB 559.4KB/s   00:04    
calculating dependencies...done.
nothing to do.
[root@cross-dev ~]# pkgin in build-essentiail
calculating dependencies...done.
nothing to do.
[root@cross-dev ~]# pkgin ls
bmake-20150505       Portable (autoconf) version of NetBSD 'make' utility
bootstrap-mk-files-20170802 *.mk files for the bootstrap bmake utility
bsdinstall-20160108  Portable version of the BSD install(1) program
bzip2-1.0.6nb1       Block-sorting file compressor
changepass-1.3.3     Simple tool to set user passwords non-interactively
coreutils-8.29       GNU basic file, shell and text manipulation utilities
curl-7.60.0          Client that groks URLs
cwrappers-20180325   pkgsrc compiler wrappers
cyrus-sasl-2.1.27    Simple Authentication and Security Layer
db4-4.8.30           Berkeley DB version 4 from Oracle
diffutils-3.6        GNU diff utilities - find the differences between files
findutils-4.6.0nb1   The GNU find, locate, updatedb, and xargs utilities
gawk-4.1.4nb1        GNU awk
gcc49-libs-4.9.4nb7  The GNU Compiler Collection (GCC) support shared libraries
gettext-lib-0.19.8.1 Internationalized Message Handling Library (libintl)
gmp-6.1.2            Library for arbitrary precision arithmetic
grep-3.1nb1          GNU grep
gsed-4.5             GNU implementation of sed, the POSIX stream editor
gtar-base-1.30       The GNU tape archiver with remote magnetic tape support
http-parser-2.8.1    Parser for HTTP messages written in C
icu-61.1             Robust and full-featured Unicode services
less-530             Pager similar to more and pg
libarchive-3.3.2     Library to read/create different archive formats
libcares-1.14.0      Asynchronous resolver library, c-ares
libestr-0.1.10       Library for some string essentials
libfastjson-0.99.8   Fast json library for C
libffi-3.2.1nb4      Foreign function interface
libiconv-1.14nb3     Character set conversion library
libidn2-2.0.4        Convert internationalized domain names to/from ASCII Encoding
liblogging-1.0.5     Easy to use logging library
liblognorm-2.0.5     Tool to normalize log data
libssh2-1.8.0        SSH2 protocol library
libunistring-0.9.10  Unicode string library
libuuid-2.31         Generate unique identifiers for objects
libuv-1.21.0         Cross-platform asychronous I/O
libxml2-2.9.8nb1     XML parser library from the GNOME project
mozilla-rootcerts-1.0.20180111 Root CA certificates from the Mozilla Project
mpfr-4.0.1           GMP-based library for multiple-precision floating-point computations
nawk-20121220nb1     Brian Kernighan's pattern-directed scanning and processing language
nbsed-20120308       NetBSD-current's sed(1)
ncurses-6.1          CRT screen handling and optimization package
nghttp2-1.32.0       Implementation of HTTP/2 in C
nodejs-10.5.0        V8 JavaScript for clients and servers
npm-6.1.0            Package manager for JavaScript
openldap-client-2.4.46 Lightweight Directory Access Protocol libraries and client programs
openssl-1.0.2o       Secure Socket Layer and cryptographic library
patch-2.7.6          Patch files using diff output
pcre-8.42            Perl Compatible Regular Expressions library
perl-5.26.2          Practical Extraction and Report Language
pkg_alternatives-1.6 Generic wrappers for programs with similar interfaces
pkg_install-20180425 Package management and administration tools for pkgsrc
pkg_install-info-4.5nb3 Standalone GNU info file installation utility
pkgin-0.11.4         Apt / yum like tool for managing pkgsrc binary packages
pkgsrc-gnupg-keys-20170418 GnuPG keys for pkgsrc infrastructure
postfix-3.3.1        Postfix SMTP server and tools
python27-2.7.15nb1   Interpreted, interactive, object-oriented programming language
readline-7.0         GNU library that can recall and edit previous input
rsyslog-8.35.0       The enhanced syslogd for Unix
smtools-20160926     Joyent tools relevant to SmartOS and SmartMachines
sqlite3-3.24.0       SQL Database Engine in a C Library
sudo-1.8.22nb1       Allow others to run commands as root
tcp_wrappers-7.6.4   Monitor and filter incoming requests for network services
wget-1.19.5          Retrieve files from the 'net via HTTP and FTP
xmlcatmgr-2.2nb1     XML and SGML catalog manager
xz-5.2.4             General-purpose data compression software
zlib-1.2.11          General purpose data compression library
zoneinit-1.6.9       Joyent Instance initialization toolchain
[root@cross-dev ~]# pkgin in build-essential
calculating dependencies...done.

1 package to refresh:
  curl-7.61.0

38 packages to install:
  libidn-1.33 p5-Net-SSLeay-1.85 p5-Net-LibIDN-0.12nb9 p5-Mozilla-CA-20180117 p5-Socket6-0.28nb1 p5-Net-IP-1.26nb5 p5-MIME-Base64-3.15nb3
  p5-IO-Socket-INET6-2.72nb3 p5-Digest-MD5-2.55nb2 mit-krb5-1.16.1nb1 p5-GSSAPI-0.28nb8 p5-Digest-HMAC-1.03nb7 p5-Net-Domain-TLD-1.75nb1
  p5-Net-DNS-1.15 p5-IO-CaptureOutput-1.11.04nb3 p5-TimeDate-2.30nb4 p5-IO-Socket-SSL-2.056 pcre2-10.31 p5-Net-SMTP-SSL-1.04nb1 p5-MailTools-2.20
  p5-Error-0.17026 p5-Email-Valid-1.202nb1 p5-Authen-SASL-2.16nb5 expat-2.2.4nb1 libtool-info-2.4.6 libtool-fortran-2.4.6nb1 libtool-base-2.4.6nb1
  m4-1.4.18 libtool-2.4.6 gmake-4.1nb4 git-docs-2.18.0 git-base-2.18.0 gcc49-4.9.4nb6 bison-3.0.4nb3 binutils-2.26.1nb1 automake-1.15.1
  autoconf-2.69nb7 build-essential-1.2

1 to refresh, 0 to upgrade, 38 to install
131M to download, 393M to install

proceed ? [Y/n] y
```

Not sure exactly what's going on here. Is there additional information I can grab to figure out why this happened? Zone has plenty of dram/cpu.

**Comment by jperkin** [comment]
Are those `build-essentiail` typos as typed?  I definitely need to add better error messages to distinguish between nothing to do due to the package already being installed, and the package not being available (as it wouldn't be in this case).

**Comment by rmustacc** [comment]
Sigh, almost certainly. Sorry for the noise here. I didn't realize it was a typo there. I guess I retyped it correctly.

**Comment by jperkin** [comment]
I'll reopen as we really should improve the output.

**Comment by jperkin** [comment]
This was fixed in joyent/pkgin@9b6b39f88304c06378d57d86235ad9fc901a2d29 and released in 0.11.5 which will be available in 2018Q3.
